workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ModalScreenOverlayDialogFactory.kt

```diff
@@ -0,0 +1,100 @@
+package com.squareup.workflow1.ui.container
+
+import android.app.Dialog
+import android.content.Context
+import android.view.KeyEvent
+import android.view.KeyEvent.ACTION_UP
+import android.view.KeyEvent.KEYCODE_BACK
+import android.view.KeyEvent.KEYCODE_ESCAPE
+import android.view.View
+import android.view.Window
+import android.view.WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL
+import com.squareup.workflow1.ui.R
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backPressedHandler
+import com.squareup.workflow1.ui.buildView
+import com.squareup.workflow1.ui.environment
+import com.squareup.workflow1.ui.showRendering
+import java.lang.IllegalStateException
+import kotlin.reflect.KClass
+
+/**
+ * Convenient base class for building [ScreenOverlay] UIs that are compatible
+ * with [View.backPressedHandler], and which honor the [ModalArea] constraint
+ * placed in the [ViewEnvironment] by the standard [BodyAndModalsScreen] container.
+ *
+ * Ironically, [Dialog] instances are created with [FLAG_NOT_TOUCH_MODAL], to ensure
+ * they do not interfere with events outside of [ModalArea].
```

Comment:
```suggestion
 * Ironically, [Dialog] instances are created with [FLAG_NOT_TOUCH_MODAL], to ensure
 * they will pass on events outside of [ModalArea] to layers below.
```

Reply:
That's misleading, though. I don't think a non-modal window is involved in handling such events, that's the window manager I'm pretty sure.

How about:

```kotlin
 * Ironically, [Dialog] instances are created with [FLAG_NOT_TOUCH_MODAL], to ensure
 * that events outside of [ModalArea] reach views in lower windows.
```